### PR TITLE
Improve attribute parsing

### LINKED ISSUE #3: [bug] accidentally converts a multiline attribute into multiple attributes
Continued from https://github.com/ryansolid/dom-expressions/issues/143

input:

```html
     <div
            multiline="
              foo
              bar
            "
          ></div>
```

output:

```js
     [
      {
        type: 'tag',
        name: 'div',
        voidElement: false,
        attrs: { foo: '', bar: '' },
        children: []
      }
    ]
```

### LINKED ISSUE #143: multiline attributes work in JSX templates, but not in `html` templates
JSX working: https://playground.solidjs.com/?hash=-570526116&version=1.4.1

`html` not working: https://playground.solidjs.com/?hash=382249257&version=1.4.1

It treats each line inside the multiline attribute as an attribute:

```html
<textarea one two three ...></textarea>
```

### PR DESCRIPTION
**Pull request by trusktr**
fixes #3
fixes https://github.com/ryansolid/dom-expressions/issues/143

After merging this one and releasing, then the [dom-expressions PR](https://github.com/ryansolid/dom-expressions/pull/146) needs to get the updated version before merging.

### CODE REVIEW

**Comment by ryansolid** [comment]
Can you tell this was compiled from CoffeeScript?

Thanks.

**Comment by trusktr** [comment]
Hahah, no, I didn't know. I thought it was just particular old-style JS code. :laughing:

File: src/parse.js [hunk 1]
```diff
@@ -1,13 +1,21 @@
 // Based on package html-parse-stringify2
 // Expanded to handle webcomponents
 
-var attrRE, lookup, parseTag, pushCommentNode, pushTextNode, tagRE;
+const tagRE = /(?:<!--[\S\s]*?-->|<(?:"[^"]*"['"]*|'[^']*'['"]*|[^'">])+>)/g;
 
-tagRE = /(?:<!--[\S\s]*?-->|<(?:"[^"]*"['"]*|'[^']*'['"]*|[^'">])+>)/g;
+// const attrRE = /\s([^'"/\s><]+?)[\s/>]|([^\s=]+)=\s?(".*?"|'.*?')/g
```
Comment (trusktr) [inline src/parse.js:6]: oops, I forgot to delete a commented line

File: src/parse.js [hunk 2]
```diff
@@ -57,43 +63,30 @@ parseTag = function (tag) {
...
 
-    if (result === null) {
-      break
-    }
-
-    if (!result[0].trim()) {
-      continue
-    }
+  // for (const match of matches) {
+  // for (const match of matchesArray) {
```
Comment (trusktr) [inline src/parse.js:70]: Strangely, this wouldn't work with a `for-of` loop. It just wouldn't iterate, as if the iterator what empty or something (is Babel messing up?). I was scratching my head for a while, then switched to a plain for loop and it worked.

**Pull request merged**